test/server/cards/01-Core/Covert.spec.js

```diff
@@ -0,0 +1,176 @@
+                expect(this.sotorii.covert).toBe(true);
+
+                this.player2.clickCard(this.sotorii);
+                this.player2.clickCard(this.masterAlchemist);
+                expect(this.game.currentConflict.defenders).toContain(this.masterAlchemist);
+                expect(this.game.currentConflict.defenders).not.toContain(this.sotorii);
+                this.player2.clickPrompt('Done');
+                expect(this.game.currentConflict.defenders).toContain(this.masterAlchemist);
+                expect(this.game.currentConflict.defenders).not.toContain(this.sotorii);
+            });
+
+            it('multiple covert', function() {
+                this.noMoreActions();
+                this.player1.clickRing('air');
+                this.player1.clickCard(this.shameful);
+                this.player1.clickCard(this.mitsu);
+                this.player1.clickCard(this.ikehata);
+                this.player1.clickCard(this.sotorii);
+                this.player1.clickCard(this.masterAlchemist);
+                this.player1.clickPrompt('Initiate Conflict');
+                expect(this.player2).toHavePrompt('Choose defenders');
+                expect(this.masterAlchemist.covert).toBe(true);
+                expect(this.sotorii.covert).toBe(true);
+
+                this.player2.clickCard(this.sotorii);
+                this.player2.clickCard(this.masterAlchemist);
+                expect(this.game.currentConflict.defenders).not.toContain(this.masterAlchemist);
+                expect(this.game.currentConflict.defenders).not.toContain(this.sotorii);
+                this.player2.clickPrompt('Done');
+                expect(this.game.currentConflict.defenders).not.toContain(this.masterAlchemist);
+                expect(this.game.currentConflict.defenders).not.toContain(this.sotorii);
+            });
+
+            it('multiple covert - reaction to covert', function() {
+                this.noMoreActions();
+                this.player1.clickRing('air');
+                this.player1.clickCard(this.shameful);
+                this.player1.clickCard(this.mitsu);
+                this.player1.clickCard(this.tengu);
+                this.player1.clickCard(this.sotorii);
+                this.player1.clickCard(this.masterAlchemist);
+                this.player1.clickPrompt('Initiate Conflict');
+                expect(this.masterAlchemist.covert).toBe(true);
+                expect(this.sotorii.covert).toBe(true);
+
+                expect(this.player1).toHavePrompt('Triggered Abilities');
+                expect(this.player1).toBeAbleToSelect(this.tengu);
+                this.player1.clickCard(this.tengu);
+
+                expect(this.player2).toHavePrompt('Choose defenders');
+
+                this.player2.clickCard(this.sotorii);
+                this.player2.clickCard(this.masterAlchemist);
+                expect(this.game.currentConflict.defenders).not.toContain(this.masterAlchemist);
```

Comment:
Who has tengu's triggered ability applied here?

Reply:
Debugging shows Alchemist.  The issue with using covert during selection is that you don't get to know who gets targeted by these effects